app/services/taxonomy/taxonomy_export.rb

```diff
@@ -3,30 +3,87 @@
 module Taxonomy
   class TaxonomyExport
     COLUMNS = %w[
-      title description content_id base_path document_type first_published_at public_updated_at
+      title
+      description
+      content_id
+      base_path
+      document_type
+      first_published_at
+      public_updated_at
     ].freeze
 
+    CSV_COLUMNS = COLUMNS + ['primary_publishing_organisation']
+
     def initialize(content_id)
       @content_id = content_id
     end
 
     def to_csv
       CSV.generate(headers: true) do |csv|
-        csv << COLUMNS
+        csv << CSV_COLUMNS
         tagged_content.each do |tagged_item|
-          csv << tagged_item.slice(*COLUMNS)
+          csv << tagged_item.slice(*CSV_COLUMNS)
         end
       end
     end
 
   private
 
     def tagged_content
-      Services.publishing_api.get_linked_items(
+      linked_items.map do |item|
+        add_primary_publishing_organisation(item)
+      end
+    end
+
+    def add_primary_publishing_organisation(item)
+      item.merge('primary_publishing_organisation' => primary_publishing_organisation_name(item['content_id']))
+    end
+
+    def linked_items
+      @linked_items ||= Services.publishing_api.get_linked_items(
         @content_id,
         link_type: 'taxons',
         fields: COLUMNS
       )
     end
+
+    def tagged_content_ids
+      linked_items.map do |content_item|
+        content_item['content_id']
+      end
+    end
+
+    def links_for_tagged_content_ids
+      links_for_content = {}
+      tagged_content_ids.each_slice(1000) do |batch_content_ids|
+        links_for_content.merge!(Services.publishing_api.get_links_for_content_ids(batch_content_ids).to_h)
+      end
+      links_for_content
+    end
```

Comment:
I think you can ask the Publishing API through `get_linked_items` to give you the links, if you ask for them. This should remove the need to ask for them here.